### PR TITLE
Improve the `watchDmChannelsTask` and the `taskScheduler`

### PR DESCRIPTION
**Pull request by SimFG**
/kind improvement
Signed-off-by: SimFG <bang.fu@zilliz.com>

### CODE REVIEW

**Comment by SimFG** [comment]
rerun ut

**Comment by SimFG** [comment]
rerun ut

File: internal/querynode/task.go [hunk 1]
```diff
@@ -152,19 +153,19 @@ func (w *watchDmChannelsTask) Execute(ctx context.Context) (err error) {
...
-
 	// check if all channels has been loaded, if YES, should do nothing and return
 	// in case of query coord trigger same watchDmChannelTask on multi
-	if len(vChannels) == loadedChannelCounter {
+	sort.Strings(vChannels)
+	sort.Strings(coll.vChannels)
+	isContains := true
+	vChannelsLen := len(coll.vChannels)
+	for i, channel := range vChannels {
+		if i >= vChannelsLen || coll.vChannels[i] != channel {
```
Comment (bigsheeper) [inline internal/querynode/task.go:163]: any additional logic here? seems harder to understand than before

**Pull request closed without merging**